clippy_lints/src/needless_update.rs

```diff
@@ -21,7 +21,14 @@ declare_clippy_lint! {
     /// #     z: i32,
     /// # }
     /// # let zero_point = Point { x: 0, y: 0, z: 0 };
-    ///
+    /// #
+    /// # #[non_exhaustive]
```

Comment:
Maybe the definition should be visible to illustrate the point?

Reply:
Hm, I don't think we need the changes to the examples at all. The fact that the lint now doesn't trigger on non exhaustive structs seems like the expected and intuitive thing to me.
Instead, maybe we can add the sentence below to the 'What it does section'?

Reply:
fixed